lib/elixir/lib/task.ex

```diff
@@ -795,9 +814,17 @@ defmodule Task do
   @doc since: "1.11.0"
   @spec await_many([t], timeout) :: [term]
   def await_many(tasks, timeout \\ 5000) when is_timeout(timeout) do
+    for task <- tasks do
+      case task do
+        %Task{} -> :ok
+        %Task.Completed{} -> :ok
+        invalid_task -> raise(%MatchError{term: invalid_task})
+      end
+    end
```

Comment:
This feels weird, but I couldn't figure out a better way to maintain the existing behaviour of "raise a MatchError if tasks contains something that's not a task". I'm open to ideas here.